ouroboros-consensus/src/Ouroboros/Consensus/Node/ProtocolInfo/Byron.hs

```diff
@@ -40,46 +48,113 @@ import qualified Test.Cardano.Chain.Genesis.Dummy as Dummy
   ProtocolInfo
 -------------------------------------------------------------------------------}
 
--- TODO This is currently configured for the demo. Parameterise so it can work
--- both for a real node and a demo node.
-protocolInfoByron :: NumCoreNodes
-                  -> CoreNodeId
-                  -> PBftParams
-                  -> Cardano.Genesis.Config
-                  -> Maybe PbftLeaderCredentials
+-- | Signature threshold. This represents the proportion of blocks in a
+--   pbftSignatureWindow-sized window which may be signed by any single key.
+newtype PBftSignatureThreshold =
+        PBftSignatureThreshold { unSignatureThreshold :: Double }
+
+-- | See chapter 4.1 of
+--   https://hydra.iohk.io/job/Cardano/cardano-ledger-specs/byronChainSpec/latest/download-by-type/doc-pdf/blockchain-spec
+defaultPBftSignatureThreshold :: PBftSignatureThreshold
+defaultPBftSignatureThreshold = PBftSignatureThreshold 0.22
+
+protocolInfoByron :: Genesis.Config
+                  -> Maybe PBftSignatureThreshold
+                  -> Update.ProtocolVersion
+                  -> Update.SoftwareVersion
+                  -> Maybe PBftLeaderCredentials
                   -> ProtocolInfo (ByronBlockOrEBB ByronConfig)
-protocolInfoByron (NumCoreNodes numCoreNodes) nid params gc mLeader =
+protocolInfoByron genesisConfig@Genesis.Config {
+                    Genesis.configGenesisHash = genesisHash
+                  , Genesis.configGenesisData =
+                      Genesis.GenesisData {
+                        Genesis.gdK                = BlockCount kParam
+                      , Genesis.gdGenesisKeyHashes = genesisKeyHashes
+                      , Genesis.gdHeavyDelegation  = genesisDelegation
+                      }
+                  }
+                  mSigThresh pVer sVer mLeader =
     ProtocolInfo {
         pInfoConfig = WithEBBNodeConfig $ EncNodeConfig {
-            encNodeConfigP = PBftNodeConfig {
-                  pbftParams   = params
-                    { -- TODO (Duncan): compute (from numGenesisKeys)
-                      pbftNumNodes = fromIntegral numCoreNodes
-                      -- Set the signature window to be short for the demo.
-                      -- TODO: Is it always 'k'?
-                      -- TODO: make pbftSignatureThreshold default to a value computed from pbftNumNodes,
-                      --       but configurable?
-                    }
-                , pbftIsLeader = proofOfCredentials nid <$> mLeader
-                }
+            encNodeConfigP   = PBftNodeConfig {
+                pbftParams          =  PBftParams
+                  { pbftSecurityParam      = SecurityParam (fromIntegral kParam)
+                  , pbftNumNodes           = fromIntegral . Set.size
+                                           . Genesis.unGenesisKeyHashes
+                                           $ genesisKeyHashes
+                  , pbftSignatureWindow    = fromIntegral kParam
```

Comment:
Are we mandating that the signing window is always `k` long? If so, should we add this to the specs?

Reply:
We probably should yes. The `cardano-sl` code interprets PBFT in such a way as it's always a K sized window. The spec leaves it as a free param. Our analysis (the 0.22) was based on the window being K. So yes I think we should fix it to K in the specs, and not make it a independent param in the impl.